extensions/jsonp/deployment/src/main/java/io/quarkus/jsonp/deployment/JsonpProcessor.java

```diff
@@ -5,16 +5,20 @@
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.builditem.FeatureBuildItem;
-import io.quarkus.deployment.builditem.nativeimage.NativeImageResourceBundleBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageResourceBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
 
 public class JsonpProcessor {
 
     @BuildStep
     void build(BuildProducer<FeatureBuildItem> feature,
             BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
-            BuildProducer<NativeImageResourceBundleBuildItem> resourceBundle) {
+            BuildProducer<NativeImageResourceBuildItem> serviceProviders) {
         reflectiveClass.produce(
                 ReflectiveClassBuildItem.builder(JsonProviderImpl.class.getName()).build());
+
+        // Register META-INF/services/jakarta.json.spi.JsonProvider as a resource, accessed by SmallRyeHealthReporter
+        serviceProviders.produce(new NativeImageResourceBuildItem("META-INF/services/jakarta.json.spi.JsonProvider"));
```

Comment:
Sorry I missed that earlier but is there a reason why you don't use `ServiceProviderBuildItem`? It seems more appropriate?

Reply:
Good point, a `ServiceProviderBuildItem` without any providers just the job and seems more appropriate.

Thanks Guillaume!